pkg/nsx/services/vpc/vpc.go

```diff
@@ -871,16 +878,15 @@ func (s *VPCService) checkVPCRealizationState(createdVpc *model.Vpc, newVpcPath
 	return nil
 }
 
-func (s *VPCService) checkLBSRealization(createdLBS *model.LBService, createdVpc *model.Vpc, nc *common.VPCNetworkConfigInfo, newVpcPath string) error {
+func (s *VPCService) checkLBSRealization(createdLBS *model.LBService, createdVpc *model.Vpc, nc *common.VPCNetworkConfigInfo, newVpcPath string) (*model.LBService, error) {
 	if createdLBS == nil {
-		return nil
+		return nil, nil
 	}
 	newLBS, err := s.NSXClient.VPCLBSClient.Get(nc.Org, nc.NSXProject, *createdVpc.Id, *createdLBS.Id)
 	if err != nil || newLBS.ConnectivityPath == nil {
 		log.Error(err, "Failed to read LBS object after creating or updating", "LBS", createdLBS.Id)
-		return err
+		return &newLBS, err
```

Comment:
&newLBS -> nil? Otherwise there will be error when err is not nil and newLBS is nil?